app/parse.py

```diff
@@ -15,8 +27,57 @@ class Product:
     num_of_reviews: int
 
 
+def get_products_from_page(driver: WebDriver, url: str) -> list[Product]:
+    driver.get(url)
+    time.sleep(1)
+    driver.refresh()
+    while True:
+        try:
+            button = driver.find_element(By.CLASS_NAME, "ecomerce-items-scroll-more")
+        except NoSuchElementException:
+            break
+        if button.get_attribute("style") == "display: none;":
+            break
+        button.click()
+        time.sleep(1)
+    products_soup = driver.find_elements(By.CLASS_NAME, "product-wrapper")
+    products = []
+    for product_soup in products_soup:
+        products.append(
+            Product(
+                title=product_soup.find_element(By.CLASS_NAME, "title").get_attribute("title"),
+                description=product_soup.find_element(By.CLASS_NAME, "description").text,
+                price=float(product_soup.find_element(By.CLASS_NAME, "price").text[1:]),
+                rating=len(product_soup.find_elements(By.CSS_SELECTOR, ".ratings span")),
+                num_of_reviews=int(product_soup.find_element(By.CLASS_NAME, "review-count").text.split()[0]),
+            )
+        )
+
+    return products
+
+
+def write_to_csv(products: list[Product], filename: str) -> None:
+    with open(filename, "w", newline="") as file:
+        writer = csv.writer(file)
+        writer.writerow(["title", "description", "price", "rating", "num_of_reviews"])
+        writer.writerows([astuple(product) for product in products])
+
+
 def get_all_products() -> None:
-    pass
+    with webdriver.Chrome() as driver:
```

Comment:
Load the homepage immediately, allowing the WebDriver to start interacting with the elements faster.
driver.get(HOME_URL)